.github/workflows/test.yml

```diff
@@ -0,0 +1,30 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - master
+      - develop
+      - feature/carthage
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: macos-12
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup
+        run: |
+          carthage update --use-xcframeworks
+          pod install
+
+      - name: Run Unit Tests
+        run: xcodebuild -scheme ZincTests -workspace Zinc-ObjC.xcworkspace -destination 'platform=iOS Simulator,name=iPhone 14' -resultBundlePath TestResults test
```

Comment:
Potential follow-up: we could consider integrating fastlane to making running the tests easier